src/glade.py

```diff
@@ -136,11 +155,100 @@ def to_strings(regex):
             with the expanded string.
             """
             expansion = ''.join(lst)
-            print("Expansion %s:\tregex:%s" % (repr(expansion), str(regex)))
+            #print("Expansion %s:\tregex:%s" % (repr(expansion), str(regex)))
             yield expansion
 
 str_db = {}
 regex_map = {}
+regex_dict = dict()
+
+def get_candidates(regex):
+    regex1 = copy.deepcopy(regex)
+    return traverse(regex1)
+
+
+def get_checks(l_curr, candidate):
+    return {}
+def check_candidate(s):
+    return True
+
+# The traverse function is the generator of candidates, it's called at each step once, it selects a terminal substring 
+# and generates all posssible generalization. Each representing a candidate regex.
+
+def traverse(regex):
+    exp = False # Used to insure that we don't modify more that one branch in each step.
+    if isinstance(regex, Rep):
+        print("It's a Rep")
+        regex.extra = 0
+        for x in traverse(regex.a):
+            print("xxxxxxx")
+            if x == -1: # -1 means we reached a leaf that is generizable.
```

Comment:
Can we use some named constant instead of just `-1`? call it `GENERLIZABLE_LEAF` perhaps?